runtime/src/fees.rs

```diff
@@ -0,0 +1,290 @@
+			let year_offset = CurrencyFeeRedemptionAdapter::<pallet_balances::Pallet<Runtime>,()>::bucketed_year_offset::<Runtime>(&0);
+
+			// Adapter no-ops if adjusted_fees is 0
+			correct_and_deposit_fee_with_passing_defaults(&acc, 2, 0);
+			assert!(
+				pallet_creditcoin::RetainedFees::<Runtime>::get(&year_offset, &acc).is_none(),
+				"0-fees are no-ops"
+			);
+
+			//Adapter inserts if fees > 0
+			correct_and_deposit_fee_with_passing_defaults(&acc, 2, 1);
+			let fee = pallet_creditcoin::RetainedFees::<Runtime>::get(&year_offset, &acc)
+				.expect("fees inserted for redemption");
+			assert_eq!(fee, 1u128);
+
+			//Adapter aggregates
+			correct_and_deposit_fee_with_passing_defaults(&acc, 2, 1);
+			let fee = pallet_creditcoin::RetainedFees::<Runtime>::get(&year_offset, &acc)
+				.expect("fees inserted for redemption");
+			assert_eq!(fee, 2u128);
+		};
+
+		ExtBuilder::build().execute_with(test);
+	}
+
+	#[test]
+	fn creditcoin_on_init_redeems() {
+		let test = || {
+			let acc = generate_account("Somebody");
+			System::set_block_number(1);
+			//offset fees by a block-year
+			let year_offset = CurrencyFeeRedemptionAdapter::<pallet_balances::Pallet<Runtime>,()>::bucketed_year_offset::<Runtime>(&0);
+
+			correct_and_deposit_fee_with_passing_defaults(&acc, 2, 1);
+
+			for block_num in 2..year_offset {
+				// simulate the runtime calling on_initialize() for each block
+				Creditcoin::on_initialize(block_num);
+
+				// ballance isn't expected to change b/c the fees haven't
+				// been released back yet
+				let free = Balances::free_balance(&acc);
+				assert_eq!(free, existential_deposit() + 1u128);
+			}
+
+			//Give back excess fees worth one.
+			Creditcoin::on_initialize(year_offset);
+			let free = Balances::free_balance(&acc);
+			assert_eq!(free, existential_deposit() + 2u128);
+
+			let event = System::events().pop().expect("FeeRedemption").event;
+
+			assert_matches!(
+				event,
+				crate::Event::Creditcoin(pallet_creditcoin::Event::<Runtime>::FeeRedemption(bn, account,amount)) => {
+					assert_eq!(bn, year_offset);
+					assert_eq!(account, acc);
+					assert_eq!(amount, 1u128);
+				}
+			);
+
+			let second_year_offset = CurrencyFeeRedemptionAdapter::<
+				pallet_balances::Pallet<Runtime>,
+				(),
+			>::bucketed_year_offset::<Runtime>(&10);
+			assert!(second_year_offset > year_offset);
+			for block_num in year_offset + 1..second_year_offset {
+				// simulate rolling to the next bucket (e.g. next year)
+				Creditcoin::on_initialize(block_num);
+
+				// ballance isn't expected to change b/c no more fees
+				// have been redeemed (b/c no other transactions)
+				let free = Balances::free_balance(&acc);
+				assert_eq!(free, existential_deposit() + 2u128);
+			}
+		};
+		ExtBuilder::build().execute_with(test);
+	}
+
+	#[test]
+	fn withdraw_fee_should_return_none_when_fee_is_zero() {
+		let test = || {
+			let who = generate_account("Somebody");
+			let call = Call::remark { remark: b"Hello World".to_vec() };
```

Comment:
@CAGS295 I can't figure out what's expected here. This snippet triggers compiler error:
```
    = note: expected reference `&Call`
               found reference `&frame_system::Call<_>`
```